core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1806,12 +1789,9 @@ class KafkaApis(val requestChannel: RequestChannel,
           createAclsRequest.getErrorResponse(requestThrottleMs,
             new SecurityDisabledException("No Authorizer is configured on the broker.")))
       case Some(auth) =>
-        val errors = mutable.HashMap[Int, Throwable]()
-        val creations = ListBuffer[(Resource, Acl)]()
-        for (i <- 0 until createAclsRequest.aclCreations.size) {
-          val result = toScala(createAclsRequest.aclCreations.get(i).acl.toFilter)
-          result match {
-            case Failure(throwable) => errors.put(i, throwable)
+        val aclCreationResults = createAclsRequest.aclCreations.asScala.map { aclCreation =>
```

Comment:
It looks good overall, but there are several issues.

One issue here is that we lost all the nice error messages about UNKNOWN and ANY, as well as the fact that this is an `InvalidRequestException` rather than an internal server error.  We ought to keep the error handling in this refactor.